File: java/spring-resilience4j/spring-resilience4j-api-b/src/main/java/com/bmarques/springresilience4japib/invoice/InvoiceController.java
package com.bmarques.springresilience4japib.invoice;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/invoice")
public class InvoiceController {

    @Autowired
    private InvoiceService service;

    @GetMapping
    public List<Invoice> findAll() {
        return service.findAll();
    }
}

File: kotlin/spring-resilience4j/spring-resilience4j-api-b/src/main/kotlin/com/bmarques/springresilience4japib/invoice/InvoiceController.kt
package com.bmarques.springresilience4japib.invoice

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/invoice")
class InvoiceController(val service: InvoiceService) {

    @GetMapping
    fun findAll(): List<Invoice> = service.findAll()
}